Tests/CADCacheAdvanceTests/CADCacheAdvanceTests.m

```diff
@@ -0,0 +1,616 @@
+//
+//  Created by Dan Federman on 4/24/20.
+//  Copyright © 2020 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS"BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#if GENERATED_XCODE_PROJECT
+// This import works when working within the generated CacheAdvance.xcodeproj used in CI.
+#import <CADCacheAdvance/CADCacheAdvance-Swift.h>
+#else
+// This import works when working within Package.swift.
+@import CADCacheAdvance;
+#endif
+@import XCTest;
+
+/// Tests that exercise the API of CADCacheAdvance.
+/// - Note: Since CADCacheAdvance is a thin wrapper on CacheAdvance<Data>, these tests are intended to do nothing more than exercise the API.
+@interface CADCacheAdvanceTests : XCTestCase
+@end
+
+@implementation CADCacheAdvanceTests
+
+// MARK: Behavior Tests
+
+- (void)test_fileURL_returnsUnderlyingFileURL;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    XCTAssertEqualObjects(cache.fileURL, [self testFileLocation]);
+}
+
+- (void)test_isWritable_returnsTrueForAWritableCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    XCTAssertTrue(cache.isWritable);
+}
+
+- (void)test_isEmpty_returnsTrueForAnEmptyCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    XCTAssertTrue(cache.isEmpty);
+}
+
+- (void)test_isEmpty_returnsFalseForANonEmptyCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    NSError *error = nil;
+    [cache appendMessage:[@"Test" dataUsingEncoding:NSUTF8StringEncoding]
```

Comment:
So just to confirm what I think I know the answer to... `Codable` is not available to Objective-C which is the root reason why we need to rework things a bit?

Reply:
It's a couple things:
1. We do not have `Codable` available to Objective-C
1. We can not expose Swift generics to Objective-C

The second point means that even if we had `Codable` in Objective-C, we'd have no way of specifying a `<T: Codable>` generic in a way that Objective-C could use.